### PR TITLE
Fixes #29: Move lib code from cmd/app/kubernets_exec.go to pkg

### LINKED ISSUE #29: [Feature] Move lib code from cmd/app/kubernets_exec.go to pkg
Currently app code and lib code is scattered in `cmd/apps/kubernetes_exec.go` and `pkg`. This is a bit misleading and should be changed.

## Expected Behaviour

All shared code should be placed under `pkg`.

## Current Behaviour

The functions will not be private, they will be public.

## Possible Solution

I suggest to move the helm related code to `pkg/helm` and the Kubernetes related code to `pkg/kubernetes`.

Based on this there are three possible ways:

1. keep the old functions in `cmd/apps/kubernetes_exec.go` as stubs and call from there the new functions.
```
import "pkg/kubernetes"
func kubectl(....) error {
   return kubernetes.Kubectl(...)
}
```

2. keep the old functions in `cmd/apps/kubernetes_exec.go` and create new ones under `pkg/kubernetes`and `pkg/helm`

3. move complete to the new structure and modify all apps at once.

I would suggest the third solution.

## Context
I try to split the command from the packages to provide a cleaner codebase for further improvments.

### PR DESCRIPTION
**Pull request by aidun**
Fixes #29: Move lib code from cmd/app/kubernets_exec.go to pkg

- Moved code to pkg/helm, pkg/config, pkg/kubernetes
- Make functions public
- Updated all references

Signed-off-by: Markus Hartmann <markush1986@gmail.com>

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## TODO
- [x] revert substitution of kubectl in comments

### CODE REVIEW

File: cmd/apps/certmanager_app.go [hunk 1]
```diff
@@ -151,7 +152,7 @@ const CertManagerInfoMsg = `# Get started with cert-manager here:
 
 # Check cert-manager's logs with:
 
-kubectl logs -n cert-manager deploy/cert-manager -f`
+Kubectl logs -n cert-manager deploy/cert-manager -f`
```
Comment (alexellis) [inline cmd/apps/certmanager_app.go]: Why is this capitalised? This won't work

File: cmd/apps/metricsserver_app.go [hunk 2]
```diff
@@ -134,11 +135,11 @@ func MakeInstallMetricsServer() *cobra.Command {
 
 const MetricsInfoMsg = `# Check pod usage
 
-kubectl top pod
+Kubectl top pod
 
 # Check node usage
 
-kubectl top node
+Kubectl top node
```
Comment (alexellis) [inline cmd/apps/metricsserver_app.go]: I think you may have done a global search and replace, which is always dangerous. Please can you revert all of these changes and be more careful with the refactoring change?

File: cmd/apps/nginx_app.go [hunk 3]
```diff
@@ -142,7 +143,7 @@ func MakeInstallNginx() *cobra.Command {
 				return err
 			}
 
-			err = kubectl("apply", "-R", "-f", outputPath)
+			err = kubernetes.Kubectl("apply", "-R", "-f", outputPath)
```
Comment (alexellis) [inline cmd/apps/nginx_app.go]: Let's alias the package as k8s please

File: pkg/kubernetes/kubectl.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	res, _ := KubectlTask("get", "nodes", `--output`, `jsonpath={range $.items[0]}{.status.nodeInfo.architecture}`)
+
+	arch := strings.TrimSpace(string(res.Stdout))
+
+	return arch
+}
+
+func KubectlTask(parts ...string) (execute.ExecResult, error) {
+	task := execute.ExecTask{
+		Command:     "Kubectl",
```
Comment (alexellis) [inline pkg/kubernetes/kubectl.go]: Again this breaks on Linux, perhaps you tested on MacOS where some commands can run case insensitive?

File: cmd/apps/metricsserver_app.go (hunk 2 above)
Comment (aidun) [reply cmd/apps/metricsserver_app.go]: Yes, I mentioned it already. I had fixed it, if the general idea where good.

**Comment by alexellis** [comment]
You "mentioned it already" where? Please get your PR fixed up and ping me when you're ready for review.

**Comment by aidun** [comment]
In the PR description. I will fix it no problem.

File: cmd/apps/certmanager_app.go (hunk 1 above)
Comment (aidun) [reply cmd/apps/certmanager_app.go]: Done

File: cmd/apps/nginx_app.go (hunk 3 above)
Comment (aidun) [reply cmd/apps/nginx_app.go]: Done

**Comment by aidun** [comment]
@alexellis fixed the issues and aliased the package. Change the PR from WIP to "Ready for review"

**Comment by aidun** [comment]
Please tell me if the change is wanted and needed. I need to make the PR and get all new changes from master, because I'm changing everything. So maybe we can synchronize it. I will get an okay and 15 minutes time to make the PR and somebody will merge it after that.

**Comment by aidun** [comment]
@alexellis I fixed the issues, rebased and added the currently added or changed apps.

**Comment by aidun** [comment]
@alexellis I think we can close this PR, because you merged an another implementation already.

**Pull request closed without merging**